### PR TITLE
feat(p2p): add chain id to enr

### PR DESCRIPTION
**Pull request by dancoombs**
Add a `chain_id` to the ENR record for additional protection from connecting to a node on the wrong network (i.e. using the wrong bootnode ENR).

### CODE REVIEW

**Review by ch4r10t33r — APPROVED** [review]
LGTM.

File: p2p-specs/p2p-interface.md [hunk 1]
```diff
@@ -572,9 +573,19 @@ The ENR MAY contain the following entries:
...
 
+#### Chain id field
+
+ENRs MUST carry a `chain_id` key containing the ID of the network the bundler is connected to. This is done to ensure connections are made with peers on the intended Ethereum network.
+
+| Key        | Value        |
+|:-----------|:-------------|
+| `chain_id` | SSZ `uint64` |
+
+Clients SHOULD connect to peers with a `chain_id` that matches local values.
```
Comment (drortirosh) [inline p2p-specs/p2p-interface.md:584]: this is actually a MUST

File: p2p-specs/p2p-interface.md [hunk 2]
```diff
@@ -572,9 +573,19 @@ The ENR MAY contain the following entries:
...
+
 #### Mempools bitfield
 
-The ENR `mempools` entry signifies the mempools subnet bitfield with the following form
+The ENR `mempool_subnets` entry signifies the mempools subnet bitfield with the following form
```
Comment (drortirosh) [inline p2p-specs/p2p-interface.md:588]: There is no definition what a "subnet" is. I thought it was defined as mempool_ID modulu 1024.

Comment (dancoombs) [reply p2p-specs/p2p-interface.md:588]: I think we can remove mempool subnets entirely. They don't have a meaning at the moment.

**Pull request closed without merging**